File: frontend/src/types/pagination.ts
import { boolean, type Infer, nullable, number, object } from "superstruct";

export const Pagination = object({
  has_next: boolean(),
  has_previous: boolean(),
  num_pages: number(),
  current_page: number(),
  next_page_number: nullable(number()),
  previous_page_number: nullable(number()),
  total_items: number(),
  per_page: number(),
});

export type Pagination = Infer<typeof Pagination>;

File: backend/tools/data_models/pagination.py
from typing import Optional

from pydantic import BaseModel


class PaginationInfo(BaseModel):
    has_next: bool
    has_previous: bool
    num_pages: int
    current_page: int
    next_page_number: Optional[int]
    previous_page_number: Optional[int]
    total_items: int
    per_page: int
